File: my-solutions/0056-merge-intervals/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key = lambda a: a[0])
        curr = intervals[0]
        res = []
        for i in range(len(intervals)):
            if intervals[i][0] <= curr[1]:
                curr[0] = min(intervals[i][0], curr[0])
                curr[1] = max(intervals[i][1], curr[1])
            else:
                res.append(curr)
                curr = intervals[i]
        
        res.append(curr)
        return res

File: my-solutions/0056-merge-intervals/solution.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int> i, vector<int> j){
            return i[0] < j[0];
        });


        vector<vector<int>> res;
        int first = 0, last = 0;

        for(int i = 0; i < intervals.size(); ++i) {
            if(i == 0) {
                first = intervals[i][0];
                last = intervals[i][1];
                continue;
            }
            if(intervals[i][0] <= last) {
                last = max(last, intervals[i][1]);
            } else {
                res.push_back({first, last});
                first = intervals[i][0];
                last = intervals[i][1];
            }
        }

        res.push_back({first, last});

        return res;
    }
};
